### PR TITLE
[EXPERIMENT][WIP][OpenVINO] Add support for LFM2-VL

### PR DESCRIPTION
**Pull request by mlukasze**
> ⚠️ AUTOMATICALLY GENERATED BY OMEGA AGENT — REQUIRES HUMAN REVIEW ⚠️
> This PR was created by an AI agent as part of automated model enablement.
> A human maintainer must review and approve it before it can be considered for merge.
> Do **NOT** merge without human review and sign-off.

# What does this PR do?

Adds OpenVINO export and inference support for `LiquidAI/LFM2.5-VL-3B` (`Lfm2VlForConditionalGeneration`, `model_type: lfm2_vl`).

The architecture is a SigLIP2-NaFlex tiled vision tower (pixel-unshuffle projector) feeding a 30-layer LFM2.5 hybrid (short-causal-conv + minority full-attention) text decoder. Base text-only LFM2 support is already merged into `main` (#1515, #1590, #1619, #1622, #1641, #1691); this PR only adds the VLM wrapper on top, reusing the existing `Lfm2OpenVINOConfig`/`Lfm2ModelPatcher` infrastructure via `get_vlm_text_generation_config` wherever possible.

This PR is adapted near-verbatim from the closed/unmerged draft huggingface/optimum-intel#1704 (which targeted the smaller `LiquidAI/LFM2-VL-450M` checkpoint and reported validated logit diff < 0.005 vs PyTorch). No structural changes were needed to scale from 450M to 3B — the vision-config fields the dummy input generator reads (`tile_size=512`, `encoder_patch_size=16`) are identical between checkpoints.

### Files changed
1. `docs/source/openvino/models.mdx` — added `LFM2-VL` doc entry.
2. `optimum/exporters/openvino/model_configs.py` — `DummyLfm2VlVisionInputGenerator` + `Lfm2VlOpenVINOConfig` (registered for `image-text-to-text`; `MAX_TRANSFORMERS_VERSION="5.4.0"` matching the existing base-LFM2 cap, since both patchers depend on `Lfm2HybridConvCache` which changed shape in transformers>=5.5).
3. `optimum/exporters/openvino/model_patcher.py` — `Lfm2VlLanguageModelPatcher` (extends `Lfm2ModelPatcher`; nested `model.model.language_model.layers` path fix via a new shared `_get_model_layers()` helper; `inputs_embeds` passthrough; lm_head weight-tying fix for transformers 5.x), `Lfm2VlImageEmbeddingModelPatcher`, `siglip2_resize_positional_embeddings_patched` (vectorizes the per-tile positional-embedding resize loop to fix a `shape_direct_ref` tracing issue), `lfm2_vl_get_image_features_patched` (vectorizes the image-token scatter/gather to fix a `fancy_indexing_mask` tracing issue).
4. `optimum/exporters/openvino/utils.py` — registered `lfm2_vl` in the VLM + SSM model lists.
5. `optimum/intel/openvino/modeling_visual_language.py` — `OVLfm2VlLanguageModel` (mamba-style explicit conv+KV cache I/O) + `_OVLfm2VlForCausalLM` (registered in `MODEL_TYPE_TO_CLS_MAPPING`); `spatial_shapes` plumbed through `forward`/`prepare_inputs_for_generation`.
6. Tests: `test_export.py`, `test_quantization.py`, `utils_tests.py` — registered `lfm2_vl` with expected submodel op/INT8-node counts (measured against our own tiny checkpoint, see below).

### Mandatory Reuse Audit (pr-quality-checklist.md Check 6)

| New symbol | Existing candidate considered | Verdict |
| :--- | :--- | :--- |
| `Lfm2VlOpenVINOConfig` | `Lfm2OpenVINOConfig` (base, text-only) | Model-specific: new class needed for the vision+projector sub-config and `image-text-to-text` task registration; dispatches back to `Lfm2OpenVINOConfig`/`get_vlm_text_generation_config` for the language-model sub-graph, no duplication of the base LM export logic. |
| `DummyLfm2VlVisionInputGenerator` | Existing `DummyVisionInputGenerator` family | Subclassed/model-specific: SigLIP2-NaFlex tiling (`tile_size`, `encoder_patch_size`, `max_tiles`) has no existing generator; extends the base pattern rather than duplicating unrelated generators. |
| `Lfm2VlLanguageModelPatcher` | `Lfm2ModelPatcher` (base, merged) | Subclassed: reuses `Lfm2ModelPatcher.__enter__`/`__exit__` and the new shared `_get_model_layers()` helper (added to the base class, backward-compatible — verified via regression test against the real base-LFM2 tiny checkpoint) rather than copy-pasting the patcher body. |
| `Lfm2VlImageEmbeddingModelPatcher` | none (SigLIP2 vision tower has no prior OV patcher in this repo) | Model-specific: no existing patcher targets SigLIP2-NaFlex; standalone class. |
| `siglip2_resize_positional_embeddings_patched`, `lfm2_vl_get_image_features_patched` | none | Model-specific tracing fixes; each vectorizes one specific data-dependent Python loop identified during patching — no equivalent utility exists elsewhere in the file to reuse. |
| `OVLfm2VlLanguageModel` | `OVModelWithEmbedForCausalLM`, `OVModelWithMambaForCausalLM` | Subclassed: reuses `OVModelWithMambaForCausalLM`'s mamba-style explicit conv+KV cache I/O plumbing (`OVCacheWithMambaStates`, `OVOutputWithMambaStates`) rather than reimplementing cache I/O bridging from scratch. |
| `_OVLfm2VlForCausalLM` | Existing `_OVFor*` VLM wrapper classes in the same file | Follows the exact same registration pattern as sibling VLM wrapper classes (e.g. the SAM/other VLM entries in `MODEL_TYPE_TO_CLS_MAPPING`); no duplicated wrapper logic. |

### Known blocker — tiny test checkpoint (same as PR #1704)

The reference tiny checkpoint id committed to `tests/openvino/utils_tests.py` (`optimum-intel-internal-testing/tiny-random-lfm2-vl`, matching PR #1704's convention) does not yet exist on the HF Hub (401/404) — this is the same blocker that appears to have stalled #1704. This agent has no HF Hub upload access, so a human maintainer needs to:
1. Run `agent-results/optimum-intel/create_tiny_lfm2_vl.py` (already generated, produces a ~4.3M-param tiny random `lfm2_vl` checkpoint, forward-pass-validated for both text-only and image+text paths).
2. Upload the output to `optimum-intel-internal-testing/tiny-random-lfm2-vl`.

Until that upload happens, the 3 registered `lfm2_vl` tests (`test_export_23_lfm2_vl`, `test_ovmodel_load_with_compressed_weights_lfm2_vl`, `test_ovmodel_load_with_uncompressed_weights_lfm2_vl`) will fail in CI with `RepositoryNotFoundError`, not because of a code defect. To validate correctness ahead of the checkpoint upload, this PR's test suite was run locally against an equivalent tiny checkpoint generated by the same script (with `MODEL_NAMES["lfm2_vl"]` temporarily pointed at the local path, then reverted before committing — no local path appears in the diff): all 3 tests passed (see `agent-results/optimum-intel/test_run.log`). A direct PyTorch-vs-OpenVINO logit comparison on that tiny checkpoint showed max abs diff ≈ 2.4e-7 (near-perfect numerical match). A regression check against the existing, real `optimum-intel-internal-testing/tiny-random-lfm2` (base text-only LFM2) checkpoint also passed, confirming the shared `Lfm2ModelPatcher._get_model_layers()` refactor did not break the sibling text-only architecture.

This is a pre-existing/anticipated environment limitation (documented in the parent ticket's architecture report), not a defect introduced by this PR.

## Installation instructions

```bash
pip install git+https://github.com/mlukasze/optimum-intel.git@enable/LiquidAI-LFM2.5-VL-3B
pip install --pre -U openvino openvino-tokenizers nncf --extra-index-url https://storage.openvinotoolkit.org/simple/wheels/nightly
```

## Exporting cmd-line

```bash
optimum-cli export openvino -m LiquidAI/LFM2.5-VL-3B --task image-text-to-text ov_lfm2_vl
```

## Inference script

```python
from transformers import AutoProcessor
from optimum.intel.openvino import OVModelForVisualCausalLM

model_id = "ov_lfm2_vl"
model = OVModelForVisualCausalLM.from_pretrained(model_id)
processor = AutoProcessor.from_pretrained(model_id)

conversation = [
    {
        "role": "user",
        "content": [
            {"type": "image", "path": "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/tasks/car.jpg"},
            {"type": "text", "text": "Describe this image in one sentence."},
        ],
    },
]
inputs = processor.apply_chat_template(
    conversation, add_generation_prompt=True, tokenize=True,
    return_dict=True, return_tensors="pt",
)
output = model.generate(**inputs, max_new_tokens=50)
print(processor.batch_decode(output, skip_special_tokens=True)[0])
```

## Before submitting

- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by mlukasze** [comment]
Fixed a hard inference crash for any spatial_shapes value other than the one used during export tracing (inference_blocked, reported by WWB accuracy validation on the real 3B model). See PR description update for full details.

File: optimum/exporters/openvino/_ov_ops.py [hunk 1]
```diff
@@ -266,3 +266,169 @@ def convert_recurrent_selective_ssm_cell(context):
...
+# including the multi-modal projector's own reshapes -- only worked for the one `spatial_shapes`
+# value used during export.
+#
+# `Lfm2VlSpatialReshapeCell` (a `torch.nn.Module`) is registered via a `ModuleExtension` for the
+# same reason as `Siglip2ResizePositionalEmbeddingsCell`, and this function is its
+# `ConversionExtension`: it reads `height`/`width` directly out of the `spatial_shapes` *tensor*
+# at runtime (via `Gather`) and builds the target shape for OpenVINO's `Reshape` op as a runtime
+# input tensor (not a compile-time attribute), so the reshape (and everything chained off its
+# output, including the projector) stays genuinely dynamic-shape-safe.
+def convert_lfm2_vl_spatial_reshape(context):
```
Comment (rkazants) [inline optimum/exporters/openvino/_ov_ops.py:394]: it is strange to have these sub-graph converted into IR in this way ModuleExtension->conversion rule. We use it rarely when we cannot patch a code using torch like GDN or Mamba that contains loop operation.
